lib/web/servers.go

```diff
@@ -90,10 +90,13 @@ func (h *Handler) getDesktopHandle(w http.ResponseWriter, r *http.Request, p htt
 
 	desktopName := p.ByName("desktopName")
 
-	windowsDesktop, err := clt.GetWindowsDesktop(r.Context(), desktopName)
+	windowsDesktops, err := clt.GetWindowsDesktopsByName(r.Context(), desktopName)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	if len(windowsDesktops) == 0 {
+		return nil, trace.Errorf("Expected one desktop, got %d", len(windowsDesktops))
```

Comment:
- You didn't expect one desktop, you expected at least one desktop.
- In this case, I think `trace.NotFound` is appropriate.